presto-main/src/test/java/io/prestosql/type/TestDateTimeOperatorsBase.java

```diff
@@ -251,6 +252,13 @@ public void testIsDistinctFrom()
         assertFunction("DATE '2013-10-27' IS DISTINCT FROM NULL", BOOLEAN, true);
     }
 
+    @Test
+    public void testDateCastFromVarchar()
+    {
+        assertFunction("DATE '2013-02-02'", DATE, toDate(new DateTime(2013, 2, 2, 0, 0, 0, 0, UTC)));
+        assertInvalidFunction("DATE '5881580-07-12'", DATE_OUT_OF_RANGE, "Out of range for Date: 5881580-07-12");
```

Comment:
What happens if we exceed long range in `DateTimeUtils parseDate`?

Reply:
It throws INVALID_CAST_ARGUMENT

Reply:
can you add a test case for this?